rebar.config

```diff
@@ -22,8 +22,7 @@
   {smtp_rfc822_parse, return_error, 2}
  ]}.
 
-{deps, [{ranch, "^1.6.2 || ^2"},
-        {hut, "1.3.0"}]}.
+{deps, [{ranch, "~> 1.6.2 or ~> 2.0"}, {hut, "~> 1.3"}]}.
```

Comment:
Surprizingly, I can't find any specification for version range syntax. Found this cheatsheet https://devhints.io/semver and this online calculator: https://jubianchi.github.io/semver-check.
None of them know about `or`, but both mention `||`.

Any ideas how rebar3 and mix calculate version ranges? I checked rebar3 code but couldn't find anything semver releted

Reply:
Ok, seems rebar3 code is here https://github.com/erlang/rebar3/blob/5e2238707402fa6d149ca186494340ce7b9ee6a2/src/rebar_packages.erl#L162.

Maybe just use `>= 1.6.2`?

Reply:
I only tested with mix and the `||` syntax did not work while `or` did...
what was wrong with ranch 1.7?

Reply:
Based on [this](https://github.com/gen-smtp/gen_smtp/pull/221#issuecomment-702076249), I think skipping over ranch 1.7 is no longer necessary now that OTP-18 support has been dropped.

Reply:
It appears npm package manager uses "||" as logical-or whereas Hex package manager (mix) uses "or".

Reply:
It appears npm package manager uses "||" as logical-or whereas Hex package manager (mix) uses "or".